ql/src/java/org/apache/hadoop/hive/ql/parse/type/TypeCheckProcFactory.java

```diff
@@ -801,6 +797,22 @@ private boolean unSafeCompareWithBigInt(TypeInfo otherTypeInfo, TypeInfo bigintC
       }
       return false;
     }
+    
+    private boolean isDecimalCharacterComparison(TypeInfo type1, TypeInfo type2) {
+      if(type1 instanceof PrimitiveTypeInfo && type2 instanceof PrimitiveTypeInfo) {
+        Set<Set<PrimitiveObjectInspector.PrimitiveCategory>> comparisons = new HashSet<>(3);
+        comparisons.add(EnumSet.of(
+            PrimitiveObjectInspector.PrimitiveCategory.DECIMAL, PrimitiveObjectInspector.PrimitiveCategory.CHAR));
+        comparisons.add(EnumSet.of(
+            PrimitiveObjectInspector.PrimitiveCategory.DECIMAL, PrimitiveObjectInspector.PrimitiveCategory.VARCHAR));
+        comparisons.add(EnumSet.of(
```

Comment:
These rules are seem to be constant. Can you please extract them?
```
private static final Set<Set<PrimitiveObjectInspector.PrimitiveCategory>> decimalCharacterComparisonRules =
  Collections.unmodifiableSet(new HashSet<>() {{add(...)...}})
```

Reply:
Fixed in https://github.com/apache/hive/pull/1780/commits/ab7432b6f8af27755acc28b2034fd2d3870c0207. Overall there is more refactoring to be done but I want to keep it in a separate issue.